### DISCUSSION TITLE
How to point alternative location of sqlite_scanner.duckdb_extension in ATTACH?

### DISCUSSION THREAD

**Discussion by rsund** [duckdb/duckdb#6313] [Q&A] [answered]
While loading extension, it is possible to use default location or point to the actual extension-file:
```
LOAD sqlite;
LOAD './non/default/location/sqlite_scanner.duckdb_extension';
```

With ATTACH the "normal" approach
```
ATTACH 'tdb.db' AS t (TYPE sqlite);
```
tries to load the extension from default location (`~/.duckdb/extensions/...`).

Is it possible to use/point some other location for the extension-file in ATTACH?

**Comment by Mause** [comment]
You can SET home_directory as documented here: https://duckdb.org/docs/sql/configuration

**Reply by rsund** [reply]
Thank you, but unfortunately this didn't seem to fully solve the problem. The `home_directory` indeed changed, but DuckDB was still adding `.duckdb/extensions//osx_amd64/sqlite_scanner.duckdb_extension` to that path. I guess I could try to copy extensions to that path, but with a self compiled DuckDB using sources from the release artefacts (instead of cloning from git repository) that path is not valid as "version signature" is missing (two slashes after extensions in the path above. Well, maybe I need to recompile whole DuckDB once more. In any case, an option for setting `custom_extensions_path` that would override the path would be very nice.

I also tried the following and it indeed seems to find the extension, but I don't know how to avoid that Binder Error, i.e. how to define storage type:
```
ATTACH 'sqlite_file.db' AS sqlite (TYPE './sqlite_scanner.duckdb_extension');
Error: Binder Error: Unrecognized storage type "./sqlite_scanner.duckdb_extension"
```

**Comment by Mytherin** [comment, marked as answer]
What if you explicitly load the custom extension before calling ATTACH?

Alternatively, why not call INSTALL on the custom extension (i.e. INSTALL ‘/path/to/ext’)

**Reply by rsund** [reply]
I tried to explicitly load the custom extension before calling ATTACH, but it didn't help. ATTACH still tries to load from the .duckdb/extensions -path and claims that it is not found:
```
LOAD './sqlite_scanner.duckdb_extension';

ATTACH 'tdb.db' AS sqlite (TYPE sqlite);
Error: IO Error: Extension "~/.duckdb/extensions//osx_amd64/sqlite_scanner.duckdb_extension" not found.
Extension "sqlite_scanner" is an existing extension.

Install it first using "INSTALL sqlite_scanner".

ATTACH 'tdb.db' AS sqlite (TYPE './sqlite_scanner.duckdb_extension');
Error: Binder Error: Unrecognized storage type "./sqlite_scanner.duckdb_extension"
```

Using INSTALL with giving path to extension works, but with one small hiccup (i.e. loading extension explicitly before ATTACH should not be done - maybe that could be easily fixed?!):
```
INSTALL './sqlite_scanner.duckdb_extension';
LOAD sqlite;
ATTACH '/Users/reijo/Dropbox/test/tdb.db' AS sqlite (TYPE sqlite);
Error: Invalid Input Error: Initialization function "sqlite_scanner_init" from file "~/.duckdb/extensions//osx_amd64/sqlite_scanner.duckdb_extension" threw an exception: "Catalog Error: Table Function with name "sqlite_scan" already exists!"
```

However, after running INSTALL once and starting CLI again, ATTACH works as expected (extension is found from the wanted path):
```
ATTACH 'tdb.db' AS sqlite (TYPE sqlite);
FROM information_schema.tables;
┌───────────────┬──────────────┬────────────┬────────────┬──────────────────────┬───┬──────────────────────┬────────────────────┬──────────┬───────────────┐
│ table_catalog │ table_schema │ table_name │ table_type │ self_referencing_c…  │ … │ user_defined_type_…  │ is_insertable_into │ is_typed │ commit_action │
│    varchar    │   varchar    │  varchar   │  varchar   │        int32         │   │        int32         │      varchar       │ varchar  │    varchar    │
├───────────────┼──────────────┼────────────┼────────────┼──────────────────────┼───┼──────────────────────┼────────────────────┼──────────┼───────────────┤
│ sqlite        │ main         │ tbl        │ BASE TABLE │                      │ … │                      │ YES                │ NO       │               │
├───────────────┴──────────────┴────────────┴────────────┴──────────────────────┴───┴──────────────────────┴────────────────────┴──────────┴───────────────┤
│ 1 rows                                                                                                                              12 columns (9 shown) │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Thank you very much for your help!
